velox/functions/prestosql/window/AggregateWindow.cpp

```diff
@@ -175,22 +179,44 @@ class AggregateWindowFunction : public exec::WindowFunction {
     bool usePreviousAggregate;
   };
 
+  bool isEmptyFrame(vector_size_t frameStart, vector_size_t frameEnd) {
+    return (frameStart == frameEnd) && (frameEnd == -1);
+  }
+
   FrameMetadata analyzeFrameValues(
       const vector_size_t* rawFrameStarts,
       const vector_size_t* rawFrameEnds,
       vector_size_t numRows) {
-    vector_size_t firstRow = rawFrameStarts[0];
+    vector_size_t i = 0;
+    while (i < numRows && rawFrameStarts[i] < 0 && rawFrameEnds[i] < 0) {
+      i++;
+    }
+    if (i) {
+      hasEmptyPrecedingFrame_ = true;
+    }
+    if (i == numRows) {
```

Comment:
In this case it seems like the result will be null for all rows, then why not optimize that path ?